### PR TITLE
Constify the Alica model

### PR DESCRIPTION
**Pull request by hendrik-skubch**
This requires a companion PR in supplementary:
https://github.com/carpe-noctem-cassel/supplementary/pull/7

- Cleans up the engine model and limits access to model elements to const versions.
- Various simplifications

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
```
[the remark this answers is not in the document]
Comment (hendrik-skubch) [reply .clang-format]: Yea, unfortunately there are two style files (ours was first :P).
afaik .clang-format is the default name by which clang-format finds the style file (I could be wrong).
We can change to the other, but we'd need to change the shell script and the commit hook.

File: alica_engine/include/engine/Types.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (hendrik-skubch) [reply alica_engine/include/engine/Types.h]: I wanted to abstract away form the implementation and indicate the semantics instead.
Yes, it clashes with RoleSet, found that out at some point.
Bag sounds pretty good to me, +1.

File: .clang-format (hunk 1 above)
[the remark this answers is not in the document]
Comment (StephanOpfer) [reply .clang-format]: Well, it was first in the alica repository. We had ours in the cnc-msl repository... :P 
Ok, we adjust the commit hook and script.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+BasedOnStyle: Google
+Standard: Cpp11
+
```
[the remark this answers is not in the document]
Comment (hendrik-skubch) [reply .clang-format]: Yes, this is tricky.

I'm fine with either, and my personal favourite is to use bracewrapping for classes, namespaces and methods, but inline braces for blocks (loops, if, ...), it strikes a balance between clarity and not thinning the code out too much.

However, the idea of bracewrapping isn't liked by some people here at all.
Can we each send someone to a thunderdome and jsut grab some popcorn?

**Comment by hendrik-skubch** [comment]
- Renamed AgentSet and friends to AgentGrp (bag is confusing the ros guys).
- replaced .clang-format with a symlink
- Made some modifications to .clang-format as a compromise. Please check / comment / scream.

**Pull request merged**